File: javascript/bit/190_Reverse_Bits.js
/**
 * @param {number} n - a positive integer
 * @return {number} - a positive integer
 */
var reverseBits = function(n) {
    let res = 0;
    for (let i=0; i<32; i++) {
        bit = (n >>> i) & 1;
        res |= (bit << (31 - i));
    }

    return res >>> 0;  // Change the sign bit to 0
};

File: python/bit/190_Reverse_Bits.py
'''
Time: O(1) (32 bits) - constant
Space: O(1)
'''

class Solution:
    def reverseBits(self, n: int) -> int:
        res = 0
        for i in range(32):
            bit = (n >> i) & 1  # (the last num of n) & 1 
            res = res | (bit << (31 - i))

        return res
